subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/IntentionsRouter.kt

```diff
@@ -6,12 +6,16 @@ import com.avito.runner.service.model.intention.Intention
 import com.avito.runner.service.model.intention.State
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.delay
+import java.util.concurrent.atomic.AtomicBoolean
 
 public class IntentionsRouter(
     private val intentionRoutings: MutableMap<String, Channel<Intention>> = mutableMapOf(),
     loggerFactory: LoggerFactory
 ) {
 
+    private val isObserving = AtomicBoolean(false)
```

Comment:
i have no idea what i'am doing here. do we even need atomic here? Are coroutines here produces actual threads?